packages/components/src/date-time/date.js

```diff
@@ -2,29 +2,82 @@
  * External dependencies
  */
 import moment from 'moment';
+import classnames from 'classnames';
+
 // react-dates doesn't tree-shake correctly, so we import from the individual
 // component here, to avoid including too much of the library
 import DayPickerSingleDateController from 'react-dates/lib/components/DayPickerSingleDateController';
 
 /**
  * WordPress dependencies
  */
-import { Component, createRef } from '@wordpress/element';
-import { isRTL } from '@wordpress/i18n';
+import { Component, createRef, useEffect, useRef } from '@wordpress/element';
+import { isRTL, _n, sprintf } from '@wordpress/i18n';
 
 /**
  * Module Constants
  */
 const TIMEZONELESS_FORMAT = 'YYYY-MM-DDTHH:mm:ss';
 
+function DatePickerDay( { day, events = [] } ) {
+	const ref = useRef();
+
+	// a11y hack to make the `There is/are n events` string
+	// available speaking for readers,
+	// re-defining the aria-label attribute.
+	// This attribute is handled by the react-dates component.
+	useEffect( () => {
+		if ( ! ref?.current?.parentNode ) {
+			return;
+		}
+
+		if ( ! events.length ) {
+			return;
+		}
+
+		const { parentNode } = ref.current;
+
+		const eventsString = sprintf(
+			/* translators: %d: number of calendar events. */
+			_n( 'There is %d event', 'There are %d events', events.length ),
+			events.length
+		);
+
+		const initAriaLabel =
+			parentNode
+				.getAttribute( 'aria-label' )
+				.replace( /\. There (?:are|is) \d+ event\s?/, '' ) +
```

Comment:
> So if I'm reading this correctly, we're likely trying to remove the string we added from the previous iteration. This won't work when the 'There is %d event', 'There are %d events' is translated and we're using a different language than English. The reason why it works now is because there are no translation strings for this new value:

It makes totally sense. 

> I'd recommend building the full string to replace, instead of doing a half replacement.

It tries to deal with a special case which could happen when:
* The date picker is opened.
* A new event is registered.

Thinking well, I'm not sure whether it could happen either since the data won't be updated asynchronously.
Going to take a look again

Reply:
Addressed here https://github.com/WordPress/gutenberg/pull/29716/commits/e87832ce884fd5bc89298fc832e79c1704f1537c